torch/library.py

```diff
@@ -39,8 +39,6 @@ def __repr__(self):
 
     def impl(self, op_name, fn, dispatch_key=''):
         if dispatch_key == '':
-            if self.dispatch_key == '':
```

Comment:
seems like it's valid to pass no dispatch key

Reply:
@bdhirsh shared that the default key is set to `CompositeImplicitAutograd` by default

Reply:
yeahh.... I dunno if we want the default to also be `CompositeImplicitAutograd` for python though.

The problem with that default for python is that the op will decompose (i.e. run the actual python code, calling the individual base ops) before hitting any transforms / backend code. What this means in practice is that `functionalize`, `__torch_dispatch__` and `AOTAUtograd` will never see the custom op that you registered - they'll just see the decomposed ops.

So I'd tentatively vote that if we want a default, it should be `CompositeExplicitAutograd`. We should probably shop that around with other people though.